ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -818,6 +803,20 @@ private static void createReplImportTasks(
         throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(tblDesc.getDatabaseName()));
       }
     }
+
+    if (table != null) {
+      if (!replicationSpec.allowReplacementInto(table)) {
+        // If the target table exists and is newer or same as current update based on repl.last.id, then just noop it.
+        return;
+      }
+    } else {
+      // If table doesn't exist, allow creating a new one only if the database state is older than the update.
+      if ((parentDb != null) && (!replicationSpec.allowReplacementInto(parentDb))) {
```

Comment:
parentDB can be null if a createDbTask is in the head of this table import tasks. In this case, we shouldn't fail and safe to assume that we go through bootstrap load.

Some of these implementations are in the base code which has allowReplacementInto and allowEventReplacementInto methods one check for current_state_id and other checks against event_id. Probably they are there to support replv1. So, will need to find appropriate names for both the above methods Please suggest if you have any good names else we will keep the same. :)